sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -201,7 +205,7 @@ private Context startTracingSpan(Method method, Context context) {
         if (!TracerProxy.isTracingEnabled() || disableTracing) {
             return context;
         }
```

Comment:
From a perf standpoint, should we split this check into two separate if clauses, so that we only look into the context if the `TracerProxy.isTracingEnabled()` state returns true?

Reply:
I believe that could be a worthwhile change.

Reply:
Make it so!